### ISSUE TITLE
SSL Issues

### ISSUE THREAD

**Issue by gbiagomba** [OJ/gobuster#194]
Hello,

I noticed if the thread count is set too high or if am running multiple gobuster scans/searches it breaks SSL on the system and in some cases it breaks my network capabilities. Normally I can fix this by doing a "service network-manager restart" and "service networking restart". But some times even that does not fix it and I have to do a full reboot. I seems if I run 2 gobuster scans with a thread count of 50 (25 each) that seems to introduce networking issues. Then I tried running this on a server with more horsepower and I noticed a similar experience happened there too. 

I figured I should inform you so you are aware, in the meantime I am testing gobuster with the default 10 threads per scan instead of 25. Lastly, I am running Kali 2 in my VM and the server is running Ubuntu 18.04

**Comment by ESBarto** [comment]
kinda happen to me, i tried this in another insolated server and its seems that it replicate as well, should check more about this.

**Comment by FireFart** [comment]
You need to adjust the threadcount to the machine you are running gobuster on. We can't fix that in code

**Comment by gbiagomba** [comment]
@FireFart that is what am saying i did tweak it and even on a beefy server i still noticed the same issues happening. Like i can see my kali vm running on my laptop but not running on the same issue when am running on an ubuntu server with 8 cores, 2 network interface cards, and some serious HP. Any who, throttling it down seems to have addressed it for now

**Comment by FireFart** [comment]
Please try with the version from the `v3.1-cleaned` again, there were a lot of changes there. Also be sure to compile it with the latest version of go

**Comment by gbiagomba** [comment]
is there a way to find what version of gobuster am running?

**Comment by gbiagomba** [comment]
@FireFart i ran `go get -u -v github.com/OJ/gobuster` so am guessing that fetched the latest version. I will try cranking her up to over 25 threads and see what happens

**Comment by FireFart** [comment]
The gobuster version is printed in the banner when running a scan.

To get the current dev branch of gobuster:
```bash
git clone --branch v3.1-cleaned https://github.com/OJ/gobuster.git && \
  cd gobuster && \
  go get . && \
  go build && \
  go install
```

**Comment by gbiagomba** [comment]
@FireFart  so it is running version 3.0.1 and i did not get any errors in that one so it looks like its fixed BUT i will upgrade to 3.1 and keep testing
